bentoml/server/api_server.py

```diff
@@ -536,52 +556,120 @@ def handle_ml_predict(self):
             'Authorization': 'Bearer ' + access_token,
             'Content-Type': 'application/ld+json'
         }
-        MLMODEL_UUID = self.ngsild_ml_model_id
+        
+        WEATHER_UUID = 'urn:ngsi-ld:WeatherObserved:Pegomas'
         URL_ENTITIES = self.ngsild_cb_url + '/ngsi-ld/v1/entities/'
+        URL_TEMPORAL = self.ngsild_cb_url + '/ngsi-ld/v1/temporal/entities/'
         AT_CONTEXT = [ self.ngsild_at_context ]
-        ATTRIBUTE_INPUT_DATA = self.ngsild_ml_model_input
-        ATTRIBUTE_OUTPUT_DATA = self.ngsild_ml_model_output
+        TARGET_ENTITY = self.ngsild_ml_model_target_entity
 
         # Get the POST data
+        # We actually don't care about it, as we actually need to get
+        # data from different sources, i.e. not only precipitation but
+        # temperature as well.
+        # This notification just acts as atrigger
         input_data_notification = request.get_json()
-
-        input_entity = input_data_notification['data'][0]['id']
-        input_data = input_data_notification['data'][0][ATTRIBUTE_INPUT_DATA]['value']
-        logger.info('input_data received from notification: %s', input_data)
-
-        # reshaping input data into a 2D array
-        input_data = np.array([input_data]).reshape(-1,1)
+        logger.info('input_data received from notification: %s', input_data_notification)
+
+        # To predict next three months of precipitation/temperature
+        # We get the current date - 90 days, then perform get temporal
+        # with 30 days aggregation to retrieve THREE values of precipitation
+        # and temperature from WeatherObserved Pegomas
+        # The ML model being used here predict the (precipitation, temperature) 
+        # of the next month from the three previous month.
+        _90DaysBefore = datetime.now(timezone.utc) - timedelta(days=90)
```

Comment:
the number of days to get data from should be configurable

Reply:
There's a lot of things that need to be configurable, this implementation is tuned to one use case (ours). I don't think we are in a position to identify what should/must be configurable, as there could be so many different use cases, in terms of how many input data, from where (which entities/properties) these input data shall be retrieved, how many output (predicted) data, where these output data should be updated, etc ...